herddb-utils/src/main/java/herddb/proto/PduCodec.java

```diff
@@ -1872,7 +1872,7 @@ private static void writeObject(ByteBuf byteBuf, Object v) {
             byteBuf.writeDouble((Float) v);
         } else if (v instanceof Short) {
             byteBuf.writeByte(TYPE_SHORT);
-            byteBuf.writeLong((Short) v);
+            byteBuf.writeShort((Short) v);
```

Comment:
this is a bug, probably no one ever used 'short' JDBC parameters

Reply:
Nice catch!